src/components/buttons/Receive/ReceiveButton.tsx

```diff
@@ -0,0 +1,84 @@
+import { ArrowDownRightIcon } from '@heroicons/react/20/solid';
+import { useCashuContext } from '@/hooks/contexts/cashuContext';
+import ReceiveButtonContent from './ReceiveButtonContent';
+import { Button, Drawer, Modal } from 'flowbite-react';
+import XMarkIcon from '@/components/icons/XMarkIcon';
+import React, { useState } from 'react';
+import { bottomSheetDrawerTheme } from '@/themes/drawerTheme';
+
+const ReceiveButton = ({ isMobile }: { isMobile: boolean }) => {
+   const [showButtonContent, setShowButtonContent] = useState(false);
+   const { activeUnit } = useCashuContext();
+
+   // const handlePaymentSuccess = (amountUsdCents: number, wallet: Wallet, quote: string) => {
+   //    handleModalClose();
+   //    addToast('Payment received!', 'success');
+   //    dispatch(
+   //       addTransaction({
+   //          type: 'lightning',
+   //          transaction: {
+   //             amount: amountUsdCents,
+   //             date: new Date().toLocaleString(),
+   //             status: TxStatus.PAID,
+   //             mint: wallet.url,
+   //             quote,
+   //             unit: activeUnit,
+   //          },
+   //       }),
+   //    );
+   // };
+
+   const handleModalClose = () => {
+      setShowButtonContent(false);
+   };
+
+   return (
+      <>
+         <Button onClick={() => setShowButtonContent(true)} className='btn-primary'>
+            <span className='text-lg'>Receive</span>{' '}
+            <ArrowDownRightIcon className='ms-2 h-5 w-5 mt-1' />
+         </Button>
+         {isMobile ? (
+            <div>
+               <Drawer
+                  open={showButtonContent}
+                  onClose={handleModalClose}
+                  position='bottom'
+                  theme={bottomSheetDrawerTheme}
+                  style={{ height: 'calc(100% - 75px)' }}
+               >
+                  <Drawer.Header
+                     title={activeUnit === 'usd' ? 'Receive $' : 'Receive Bitcoin'}
+                     titleIcon={() => null}
+                     closeIcon={() => <XMarkIcon className='h-8 w-8' />}
+                  />
+                  <Drawer.Items className='flex flex-col h-full   '>
+                     {showButtonContent && (
+                        <ReceiveButtonContent
+                           isMobile={isMobile}
+                           closeParentComponent={handleModalClose}
+                        />
+                     )}
+                  </Drawer.Items>
+               </Drawer>
+            </div>
+         ) : (
+            <Modal show={showButtonContent} onClose={handleModalClose} size={'sm'}>
+               <Modal.Header>{activeUnit === 'usd' ? 'Receive $' : 'Receive Bitcoin'}</Modal.Header>
+               <Modal.Body>
+                  <div className='flex flex-col space-y-20 items-stretch justify-center'>
+                     {showButtonContent && (
```

Comment:
Because the modal still exists, just hidden and I want the component to be removed from the dom. It's the only way I can figure out how to get the state to reset when the modal closes

Reply:
fixed here [f2c0983](https://github.com/MakePrisms/boardwalkcash/pull/130/commits/f2c0983ebd5fbfb91eecbfe530ec65ff0fc34928)